config/initializers/devise.rb

```diff
@@ -36,6 +36,10 @@
   #
   # Uses the DevAuth strategy if local and ENV["DEV_AUTH_ENABLED"] is true
 
+  if !HostEnv.local? && FeatureFlags.dev_auth.enabled?
+    raise "The DevAuth strategy must not be used in this environment"
```

Comment:
Wondering if it is worth having a specific exeption class for this as well? Might be easier to spot - small thing though.

Reply:
Depends how much faith we have in `HostEnv.local?` being set correctly? I don't think there are many ways of checking whether an environment is considered local or not. Perhaps a check somewhere on the incoming HTTP Host name - if it looks remotely unproduction-link then flash up a warning/exception somewhere asap? i don't think there's a way of knowing what the full server URL is on system initialize?

Reply:
@willmcb the error will appear in the build history and we won't be doing anything else with it so I think just setting the message is fine.